tests/test_flow/fakes.py

```diff
@@ -177,7 +212,13 @@ def create_aip_job(body, parent):
         # Emulate this behavior by setting the log level; this should filter out
         # all (or almost all) the logs. Tests that check the log output will
         # only test against logs which are printed locally.
-        with caplog.at_level(logging.CRITICAL):
+        #
+        # Additionally, AIP executions run in an isolated environment and do not
+        # have access to disk cache from local bionic instance. It must retrieve
+        # the data from GCS. Emulate this behavior by temporarily removing all
+        # files in the tmp_path directory. The files are restored after the
+        # AIP computation is completed.
+        with caplog.at_level(logging.CRITICAL), temporary_clean_dir(tmp_path):
             run_aip(body["trainingInput"]["args"][3])
         return Mock()
```

Comment:
Does the AIP mocks here use the `FakeAipExecutor` (and other related fakes)? If it does not, should it?

Reply:
No, it is not using the fake executor. I think faking / mocking AIP API would allow us to give the most test coverage to our code. Right now the fake AIP works by replacing a global variable; I think using injection would make things more robust. Can look into this in a future PR if needed.